// @oclc/dist/cjs/themes/components/appbar.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiAppBar = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiAppBar = {
    variants: [
        {
            props: { variant: 'alt' },
            style: {
                background: base_tokens_json_1.default.tokens.palette.background.white,
                color: base_tokens_json_1.default.tokens.palette.background.blue700,
                fontWeight: 600
            }
        }
    ],
    styleOverrides: {
        root: {
            background: base_tokens_json_1.default.tokens.palette.blue.blue700,
            color: base_tokens_json_1.default.tokens.palette.background.white,
            fontWeight: 600,
            '& .MuiButtonBase-root': {
                '&.MuiIconButton-root': {
                    color: base_tokens_json_1.default.tokens.palette.background.white,
                    '&:hover': {
                        color: base_tokens_json_1.default.tokens.palette.background.white
                    }
                }
            },
            '& .MuiToolbar-root .Mui-focusVisible': {
                outlineColor: base_tokens_json_1.default.tokens.palette.background.white
            }
        },
        colorPrimary: {
            background: base_tokens_json_1.default.tokens.palette.blue.blue800
        }
    }
};
exports.default = exports.MuiAppBar;
