// @oclc/dist/cjs/themes/components/buttons.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiButtonGroup = exports.MuiIconButton = exports.MuiButton = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiButton = {
    defaultProps: {
        disableFocusRipple: true
    },
    styleOverrides: {
        root: {
            fontWeight: 600,
            borderRadius: '6px',
            minWidth: 'unset',
            textTransform: 'none',
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            },
            '&:hover, &.Mui-focusVisible': {
                '& .MuiAvatar-colorDefault': {
                    background: base_tokens_json_1.default.tokens.palette.blue.blue800
                },
                '& .MuiAvatar-alt': {
                    background: base_tokens_json_1.default.tokens.palette.blue.blue100
                }
            },
            '& .MuiButton-startIcon': {
                marginLeft: -2,
                marginRight: 4
            }
        },
        contained: {
            boxShadow: "none",
            '&.Mui-disabled': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    color: base_tokens_json_1.default.tokens.palette.background.white
                }
            },
            '&[aria-disabled="true"]': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    color: base_tokens_json_1.default.tokens.palette.background.white
                }
            }
        },
        containedPrimary: {
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue800,
                outlineOffset: '2px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            }
        },
        outlinedPrimary: {
            background: base_tokens_json_1.default.tokens.palette.background.white,
            borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue200,
                color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                outlineOffset: '3px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                borderColor: base_tokens_json_1.default.tokens.palette.blue.blue800
            }
        },
        textPrimary: {
            '&:hover, &.Mui-focusVisible': {
                color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                '& .MuiAvatar-colorDefault': {
                    background: base_tokens_json_1.default.tokens.palette.blue.blue800
                },
                '& .MuiAvatar-alt': {
                    background: base_tokens_json_1.default.tokens.palette.blue.blue100
                }
            },
            '&[aria-expanded="true"]': {
                color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                '& .MuiAvatar-colorDefault': {
                    background: base_tokens_json_1.default.tokens.palette.blue.blue800
                },
                '& .MuiAvatar-alt': {
                    background: base_tokens_json_1.default.tokens.palette.blue.blue100
                }
            }
        },
        outlined: {
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue200,
                color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                borderColor: base_tokens_json_1.default.tokens.palette.blue.blue800
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
                }
            },
            '&[aria-disabled="true"]': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
                }
            }
        },
        text: {
            color: base_tokens_json_1.default.tokens.palette.blue.blue700,
            '&[aria-disabled="true"]': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    backgroundColor: 'transparent'
                }
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    backgroundColor: 'transparent'
                }
            },
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue200,
                color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                outlineOffset: '3px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            },
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue200
            }
        },
        sizeSmall: ({ theme }) => ({
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 8,
            paddingRight: 8,
            '&.MuiButton-outlined': {
                paddingTop: `calc(${theme.spacing(1)} - 1px)`,
                paddingBottom: `calc(${theme.spacing(1)} - 1px)`,
                paddingLeft: 7,
                paddingRight: 7
            },
            ...theme.typography.labelSm,
            '& .label': base_tokens_json_1.default.tokens.typography.label,
            '& .MuiButton-startIcon': {
                '*:nth-of-type(1)': {
                    fontSize: 16
                }
            },
            '& .MuiButton-endIcon': {
                '*:nth-of-type(1)': {
                    fontSize: 16
                }
            }
        }),
        sizeMedium: ({ theme }) => ({
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 8,
            paddingRight: 8,
            '&.MuiButton-outlined': {
                paddingTop: `calc(${theme.spacing(1)} - 1px)`,
                paddingBottom: `calc(${theme.spacing(1)} - 1px)`,
                paddingLeft: 7,
                paddingRight: 7
            },
            ...theme.typography.label,
            '& .MuiButton-startIcon': {
                '*:nth-of-type(1)': {
                    fontSize: 18
                }
            },
            '& .MuiButton-endIcon': {
                '*:nth-of-type(1)': {
                    fontSize: 18
                }
            }
        }),
        sizeLarge: ({ theme }) => ({
            paddingTop: theme.spacing(1.5),
            paddingBottom: theme.spacing(1.5),
            paddingLeft: 12,
            paddingRight: 12,
            '&.MuiButton-outlined': {
                paddingTop: `calc(${theme.spacing(1.5)} - 1px)`,
                paddingBottom: `calc(${theme.spacing(1.5)} - 1px)`,
                paddingLeft: 11,
                paddingRight: 11
            },
            ...theme.typography.labelLg,
            '& .label': base_tokens_json_1.default.tokens.typography.label,
            '& .MuiButton-startIcon': {
                '*:nth-of-type(1)': {
                    fontSize: 20
                }
            },
            '& .MuiButton-endIcon': {
                '*:nth-of-type(1)': {
                    fontSize: 20
                }
            }
        })
    }
};
exports.MuiIconButton = {
    defaultProps: {
        disableFocusRipple: true,
        disableRipple: true
    },
    variants: [
        {
            props: { variant: 'outlined' },
            style: ({ theme }) => ({
                padding: `calc(${theme.spacing(1)} - 1px)`,
                border: `1px solid ${base_tokens_json_1.default.tokens.palette.blue.blue700}`,
                borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                color: base_tokens_json_1.default.tokens.palette.blue.blue700,
                background: base_tokens_json_1.default.tokens.palette.background.white,
                '&:hover, &.Mui-focusVisible': {
                    color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                    border: `1px solid ${base_tokens_json_1.default.tokens.palette.blue.blue800}`,
                    background: base_tokens_json_1.default.tokens.palette.blue.blue200
                },
                '&.Mui-disabled': {
                    background: 'transparent',
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '&.MuiIconButton-sizeLarge': {
                    padding: `calc(${theme.spacing(1.5)} - 1px)`
                }
            })
        }
    ],
    styleOverrides: {
        root: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.blue.blue800,
            padding: theme.spacing(1),
            borderRadius: 6,
            '&:hover, &.Mui-focusVisible': {
                background: base_tokens_json_1.default.tokens.palette.blue.blue200,
                color: base_tokens_json_1.default.tokens.palette.blue.blue800
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                background: 'transparent',
                pointerEvents: 'unset',
                cursor: 'not-allowed'
            },
            '&.Mui-focusVisible': {
                outline: `2px solid ${base_tokens_json_1.default.tokens.palette.text.text300}`,
                outlineOffset: '2px',
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            }
        }),
        sizeLarge: ({ theme }) => ({
            padding: theme.spacing(1.5)
        }),
        colorPrimary: {
            background: base_tokens_json_1.default.tokens.palette.blue.blue700,
            color: base_tokens_json_1.default.tokens.palette.background.white,
            '& .MuiSvgIcon-root': {
                color: base_tokens_json_1.default.tokens.palette.background.white
            },
            '&:hover, &.Mui-focusVisible': {
                background: base_tokens_json_1.default.tokens.palette.blue.blue800
            },
            '&.Mui-disabled': {
                background: base_tokens_json_1.default.tokens.palette.grey.disabled,
                color: base_tokens_json_1.default.tokens.palette.background.white
            }
        }
    }
};
exports.MuiButtonGroup = {
    defaultProps: {
        disableElevation: true,
        disableFocusRipple: true
    },
    styleOverrides: {
        root: ({ theme }) => ({
            '& .MuiButton-sizeSmall:first-of-type': {
                '&.MuiButtonGroup-groupedHorizontal': {
                    borderRight: '0px'
                },
                '&.MuiButtonGroup-groupedVertical': {
                    borderBottom: '0px'
                },
                '&.Mui-focusVisible': {
                    zIndex: 1
                }
            },
            '& .MuiButton-sizeMedium:first-of-type': {
                '&.MuiButtonGroup-groupedHorizontal': {
                    borderRight: '0px'
                },
                '&.MuiButtonGroup-groupedVertical': {
                    borderBottom: '0px'
                },
                '&.Mui-focusVisible': {
                    zIndex: 1
                }
            },
            '& .MuiButton-sizeLarge:first-of-type': {
                '&.MuiButtonGroup-groupedHorizontal': {
                    borderRight: '0px'
                },
                '&.MuiButtonGroup-groupedVertical': {
                    borderBottom: '0px'
                },
                '&.Mui-focusVisible': {
                    zIndex: 1
                }
            },
            '& .MuiButton-root:last-child': {
                '&.MuiButton-contained': {
                    '&.MuiButton-sizeSmall': {
                        paddingTop: theme.spacing(1),
                        paddingBottom: theme.spacing(1),
                        paddingLeft: 4,
                        paddingRight: 4
                    },
                    '&.MuiButton-sizeMedium': {
                        paddingTop: theme.spacing(1),
                        paddingBottom: theme.spacing(1),
                        paddingLeft: 4,
                        paddingRight: 4
                    },
                    '&.MuiButton-sizeLarge': {
                        paddingTop: theme.spacing(1.5),
                        paddingBottom: theme.spacing(1.5),
                        paddingLeft: 5,
                        paddingRight: 5
                    },
                    '&:hover': {
                        backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue800
                    },
                    '&.Mui-disabled': {
                        '&:hover': {
                            backgroundColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                            color: base_tokens_json_1.default.tokens.palette.text.text300
                        }
                    }
                },
                '&.MuiButton-outlined': {
                    '&.MuiButton-sizeSmall': {
                        paddingTop: `calc(${theme.spacing(1)} - 1px)`,
                        paddingBottom: `calc(${theme.spacing(1)} - 1px)`,
                        paddingLeft: 3,
                        paddingRight: 3
                    },
                    '&.MuiButton-sizeMedium': {
                        paddingTop: `calc(${theme.spacing(1)} - 1px)`,
                        paddingBottom: `calc(${theme.spacing(1)} - 1px)`,
                        paddingLeft: 3,
                        paddingRight: 3
                    },
                    '&.MuiButton-sizeLarge': {
                        paddingTop: `calc(${theme.spacing(1.5)} - 1px)`,
                        paddingBottom: `calc(${theme.spacing(1.5)} - 1px)`,
                        paddingLeft: 4,
                        paddingRight: 4
                    },
                    '&.Mui-disabled': {
                        '&:hover': {
                            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
                            color: base_tokens_json_1.default.tokens.palette.grey.disabled
                        }
                    }
                }
            }
        }),
        contained: {
            '& .MuiButton-root': {
                '&.Mui-disabled': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    '&:hover': {
                        color: base_tokens_json_1.default.tokens.palette.text.text300
                    }
                }
            }
        }
    }
};
exports.default = exports.MuiButton;
