// @oclc/dist/cjs/themes/components/card.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiCardActionArea = exports.MuiCardHeader = exports.MuiCardMedia = exports.MuiCardActions = exports.MuiCardContent = exports.MuiCard = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiCard = {
    variants: [
        {
            props: { variant: 'callout' },
            style: {
                background: base_tokens_json_1.default.tokens.palette.grey.grey100,
                border: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey200}`,
                alignItems: 'flex-start',
                boxShadow: 'none',
                '& .MuiCardHeader-avatar': {
                    marginRight: 8
                },
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.blue.blue700
                }
            }
        },
        {
            props: { variant: 'callout', color: 'primary' },
            style: {
                background: base_tokens_json_1.default.tokens.palette.blue.blue100,
                border: `1px solid ${base_tokens_json_1.default.tokens.palette.blue.blue200}`,
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.blue.blue700
                },
                '& .MuiCardHeader-avatar': {
                    marginRight: 8
                }
            }
        },
        {
            props: { variant: 'callout', color: 'info' },
            style: {
                background: base_tokens_json_1.default.tokens.palette.purple.purple100,
                border: `1px solid ${base_tokens_json_1.default.tokens.palette.purple.purple200}`,
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.purple.purple500
                },
                '& .MuiCardHeader-avatar': {
                    marginRight: 8
                }
            }
        }
    ],
    styleOverrides: {
        root: {
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            boxShadow: '0px 1px 5px 0px #0000001F, 0px 2px 2px 0px #00000024, 0px 3px 1px -2px #00000033',
            '&.MuiPaper-outlined': {
                border: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey300}`,
                boxShadow: 'none'
            },
            '&.MuiPaper-callout': {
                '& ul': {
                    paddingInlineStart: 56,
                    '&:last-of-type': {
                        marginBottom: 0
                    }
                }
            }
        }
    }
};
exports.MuiCardContent = {
    styleOverrides: {
        root: ({ theme }) => ({
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(2),
            paddingLeft: 12,
            paddingRight: 12,
            '&:last-of-type': {
                paddingBottom: theme.spacing(2)
            }
        })
    }
};
exports.MuiCardActions = {
    defaultProps: {
        disableSpacing: true
    },
    styleOverrides: {
        root: ({ theme }) => ({
            background: base_tokens_json_1.default.tokens.palette.background.linen,
            borderTop: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey200}`,
            paddingTop: theme.spacing(2),
            paddingBottom: theme.spacing(2),
            paddingLeft: 8,
            paddingRight: 8
        })
    }
};
exports.MuiCardMedia = {
    styleOverrides: {
        root: {
            backgroundRepeat: 'no-repeat'
        }
    }
};
exports.MuiCardHeader = {
    styleOverrides: {
        root: ({ theme }) => ({
            paddingTop: theme.spacing(2),
            paddingBottom: theme.spacing(2),
            paddingLeft: 12,
            paddingRight: 12,
            '& + .MuiCardContent-root': {
                paddingTop: 0
            }
        }),
        content: {
            fontWeight: 600,
            color: base_tokens_json_1.default.tokens.palette.text.text300
        }
    }
};
exports.MuiCardActionArea = {
    styleOverrides: {
        root: {
            '&.Mui-focusVisible': {
                outlineOffset: '-2px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                overflow: 'visible'
            }
        }
    }
};
