// @oclc/dist/cjs/themes/components/checkbox.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const MuiCheckbox = {
    defaultProps: {
        disableRipple: true
    },
    styleOverrides: {
        root: {
            padding: 0,
            color: base_tokens_json_1.default.tokens.palette.grey.grey500,
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    outline: 'none'
                }
            },
            '&.Mui-checked': {
                color: base_tokens_json_1.default.tokens.palette.blue.blue700,
                '&.Mui-disabled': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled
                }
            },
            '&:hover': {
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                borderRadius: '4px',
                WebkitBorderRadius: '4px'
            },
            '&.Mui-focusVisible': {
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineOffset: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                borderRadius: '4px',
                WebkitBorderRadius: '4px'
            },
            '& .MuiSvgIcon-root': {
                fontSize: 20
            }
        },
        sizeSmall: {
            '& .MuiSvgIcon-root': {
                fontSize: 18
            }
        }
    }
};
exports.default = MuiCheckbox;
