// @oclc/dist/cjs/themes/components/chip.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const MuiChip = {
    defaultProps: {
        variant: 'outlined',
        size: 'small'
    },
    styleOverrides: {
        filled: {
            backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue100,
            border: `0.5px solid ${base_tokens_json_1.default.tokens.palette.blue.blue600}`,
            '& .MuiChip-label': {
                color: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue700
            },
            '& .MuiChip-icon': {
                color: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue700
            },
            '&:hover': {
                background: 'linear-gradient(0deg, rgba(0, 0, 0, 0.04), rgba(0, 0, 0, 0.04)), linear-gradient(0deg, #E6EFF5, #E6EFF5)'
            },
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue100
            }
        },
        sizeSmall: ({ theme }) => ({
            paddingTop: 0,
            paddingBottom: 0,
            paddingLeft: 6,
            paddingRight: 6,
            '& .MuiChip-labelSmall': {
                padding: 0,
                ...theme.typography.body2
            },
            '& .MuiChip-icon': {
                marginLeft: '-2px',
                marginRight: '4px',
                ...theme.typography.body2
            },
            '& .MuiChip-deleteIcon': {
                fontSize: '16px',
                marginLeft: '6px',
                marginRight: '-2px',
                color: base_tokens_json_1.default.tokens.palette.text.text100
            }
        }),
        sizeMedium: ({ theme }) => ({
            paddingTop: theme.spacing(0.75),
            paddingBottom: theme.spacing(0.75),
            paddingLeft: 8,
            paddingRight: 4,
            '& .MuiChip-labelMedium': {
                padding: 0,
                ...theme.typography.body2
            },
            '& .MuiChip-icon': {
                marginLeft: '-2px',
                marginRight: '4px',
                fontSize: '16px'
            },
            '& .MuiChip-deleteIcon': {
                fontSize: '16px',
                marginLeft: '8px',
                marginRight: '-1px',
                color: base_tokens_json_1.default.tokens.palette.text.text100
            }
        }),
        root: {
            height: 'unset',
            padding: 0,
            borderColor: base_tokens_json_1.default.tokens.palette.grey.grey300,
            '&.Mui-focusVisible': {
                outlineOffset: '2px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                '&:not(.MuiChip-filled)': {
                    background: 'linear-gradient(0deg, rgba(0, 0, 0, 0.04), rgba(0, 0, 0, 0.04)), #FFFFFF'
                }
            },
            '&.MuiChip-clickable': {
                '&:not(.MuiChip-filled):hover': {
                    background: 'linear-gradient(0deg, rgba(0, 0, 0, 0.04), rgba(0, 0, 0, 0.04)), #FFFFFF'
                }
            }
        }
    }
};
exports.default = MuiChip;
