// @oclc/dist/cjs/themes/components/datepicker-v6.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiMultiSectionDigitalClock = exports.MuiPickersToolbar = exports.MuiDatePickerToolbar = exports.MuiPickersLayout = exports.MuiDateRangePickerDay = exports.MuiMultiInputDateRangeField = exports.MuiPickersDay = exports.MuiDayCalendar = exports.MuiPickersMonth = exports.MuiPickersArrowSwitcher = exports.MuiPickersYear = exports.MuiYearCalendar = exports.MuiYearPicker = exports.MuiPickersCalendarHeader = exports.MuiDatePicker = exports.MuiDateTimePicker = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const mdi_material_ui_1 = require("mdi-material-ui");
exports.MuiDateTimePicker = {
    defaultProps: {
        components: {
            SwitchViewIcon: mdi_material_ui_1.ChevronDown
        },
        closeOnSelect: false
    }
};
exports.MuiDatePicker = {
    defaultProps: {
        components: {
            SwitchViewIcon: mdi_material_ui_1.ChevronDown
        }
    }
};
exports.MuiPickersCalendarHeader = {
    defaultProps: {
        components: {
            SwitchViewIcon: mdi_material_ui_1.ChevronDown
        }
    },
    styleOverrides: {
        root: ({ theme }) => ({
            marginTop: theme.spacing(4),
            marginBottom: 2,
            paddingLeft: theme.spacing(6),
            paddingRight: theme.spacing(6)
        }),
        label: {
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            fontWeight: 600
        },
        labelContainer: {
            overflow: 'visible'
        },
        switchViewButton: {
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100
            },
            '& .MuiPickersCalendarHeader-switchViewIcon': {
                color: base_tokens_json_1.default.tokens.palette.text.text200
            }
        }
    }
};
exports.MuiYearPicker = {
    styleOverrides: {
        root: {
            '& .PrivatePickersYear-root': {
                '& .PrivatePickersYear-yearButton': {
                    background: 'transparent',
                    fontSize: base_tokens_json_1.default.tokens.typography.body.fontSize,
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    '&.Mui-selected': {
                        backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                    },
                    '&:focus': {
                        backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                        outline: '2px solid black',
                        outlineOffset: '1px',
                        '&.Mui-selected': {
                            backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                        }
                    }
                }
            }
        }
    }
};
exports.MuiYearCalendar = {
    styleOverrides: {
        root: {
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
        }
    }
};
exports.MuiPickersYear = {
    styleOverrides: {
        yearButton: {
            borderRadius: 24,
            padding: 6,
            '&.Mui-selected': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                fontWeight: 600,
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '&:focus': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                outline: '2px solid black',
                outlineOffset: '1px',
                '&.Mui-selected': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                    color: base_tokens_json_1.default.tokens.palette.background.white,
                    '&:hover': {
                        backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                        color: base_tokens_json_1.default.tokens.palette.background.white
                    }
                }
            },
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                color: base_tokens_json_1.default.tokens.palette.text.text300
            },
            '&:Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            }
        }
    }
};
exports.MuiPickersArrowSwitcher = {
    styleOverrides: {
        root: {
            '& .MuiPickersArrowSwitcher-button': {
                fontSize: 18,
                padding: 4,
                margin: 0,
                color: base_tokens_json_1.default.tokens.palette.text.text200,
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100
                }
            }
        }
    }
};
exports.MuiPickersMonth = {
    styleOverrides: {
        monthButton: {
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            '&.Mui-selected': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                fontWeight: 600,
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '&:focus': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                outline: '2px solid black',
                outlineOffset: '1px',
                '&.Mui-selected': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                    color: base_tokens_json_1.default.tokens.palette.background.white,
                    '&:hover': {
                        backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                        color: base_tokens_json_1.default.tokens.palette.background.white
                    }
                }
            },
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                color: base_tokens_json_1.default.tokens.palette.text.text300
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            }
        }
    }
};
exports.MuiDayCalendar = {
    styleOverrides: {
        header: {
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
        },
        slideTransition: {
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
        },
        weekDayLabel: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.text.text100,
            ...theme.typography.caption
        }),
        monthContainer: ({ theme }) => ({
            paddingTop: theme.spacing(2),
            paddingBottom: theme.spacing(2)
        })
    }
};
exports.MuiPickersDay = {
    styleOverrides: {
        root: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            ...theme.typography.body1,
            '&.Mui-disabled': {
                color: `${base_tokens_json_1.default.tokens.palette.grey.disabled} !important`
            },
            '&.Mui-selected': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '&:focus': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                outline: '2px solid black',
                outlineOffset: '1px',
                '&.Mui-selected': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                    color: base_tokens_json_1.default.tokens.palette.background.white,
                    '&:hover': {
                        backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                        color: base_tokens_json_1.default.tokens.palette.background.white
                    }
                }
            },
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100
            }
        }),
        today: {
            borderColor: `${base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300} !important`
        }
    }
};
exports.MuiMultiInputDateRangeField = {
    styleOverrides: {
        root: {
            alignItems: 'center'
        }
    }
};
exports.MuiDateRangePickerDay = {
    styleOverrides: {
        root: {
            '&:last-of-type': {
                '& .MuiDateRangePickerDay-rangeIntervalDayPreview': {
                    borderRightColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300
                }
            },
            '&:first-of-type': {
                '& .MuiDateRangePickerDay-rangeIntervalDayPreview': {
                    borderLeftColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300
                }
            }
        },
        rangeIntervalDayPreview: {
            borderTopColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300,
            borderBottomColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300
        },
        rangeIntervalDayHighlight: {
            backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue100
        },
        day: {
            '&.Mui-selected': {
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    outline: '2px solid black'
                }
            },
            '&.Mui-disabled': {
                color: `${base_tokens_json_1.default.tokens.palette.grey.disabled} !important`
            }
        },
        dayInsideRangeInterval: {
            '&&:hover': {
                outline: '2px solid black',
                color: `${base_tokens_json_1.default.tokens.palette.text.text300} !important`
            }
        },
        outsideCurrentMonth: {
            backgroundColor: 'transparent',
            borderColor: 'transparent',
            '& .MuiDateRangePickerDay-rangeIntervalDayPreview': {
                border: 'none'
            }
        },
        lastVisibleCell: {
            '& .MuiDateRangePickerDay-rangeIntervalDayPreview': {
                borderRightColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300
            }
        },
        firstVisibleCell: {
            '& .MuiDateRangePickerDay-rangeIntervalDayPreview': {
                borderLeftColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue300
            }
        }
    }
};
exports.MuiPickersLayout = {
    styleOverrides: {
        root: {
            '& .MuiDialogActions-root': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
            },
            '& .MuiPickersLayout-actionBar': {
                padding: 12
            }
        }
    }
};
exports.MuiDatePickerToolbar = {
    defaultProps: {
        toolbarPlaceholder: '- -'
    }
};
exports.MuiPickersToolbar = {
    styleOverrides: {
        root: ({ theme }) => ({
            '& .MuiTypography-overline': {
                ...theme.typography.body,
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                textTransform: 'none'
            },
            '&.MuiDatePickerToolbar-root': {
                padding: theme.spacing(1, 3)
            },
            '&.MuiDateRangePickerToolbar-root': {
                padding: theme.spacing(1, 3)
            }
        }),
        content: ({ theme }) => ({
            '& .MuiTypography-root': {
                ...theme.typography.labelLg,
                color: base_tokens_json_1.default.tokens.palette.text.text300
            }
        })
    }
};
exports.MuiMultiSectionDigitalClock = {
    styleOverrides: {
        root: {
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            '& .MuiList-root': {
                '& > li': {
                    fontWeight: 400,
                    borderRadius: '6px',
                    '&:hover': {
                        backgroundColor: `${base_tokens_json_1.default.tokens.palette.blue.blue100} !important`
                    },
                    '&.Mui-focusVisible': {
                        backgroundColor: `${base_tokens_json_1.default.tokens.palette.blue.blue100} !important`,
                        outline: '2px solid black',
                        borderRadius: '6px',
                        WebkitBorderRadius: '6px'
                    },
                    '&.Mui-selected:not(.Mui-disabled)': {
                        color: `${base_tokens_json_1.default.tokens.palette.background.white} !important`,
                        backgroundColor: `${base_tokens_json_1.default.tokens.palette.blue.blue700} !important`
                    },
                    '&.Mui-disabled': {
                        color: `${base_tokens_json_1.default.tokens.palette.grey.disabled} !important`
                    }
                }
            }
        }
    }
};
