// @oclc/dist/cjs/themes/components/drawer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiDrawer = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const styles_1 = require("@mui/material/styles");
let theme = (0, styles_1.createTheme)();
exports.MuiDrawer = {
    styleOverrides: {
        root: {
            overflow: 'hidden',
            paddingTop: 64,
            paddingBottom: 128,
            minWidth: 240,
            '&& .MuiDrawer-paper': {
                [theme.breakpoints.down('sm')]: {
                    width: '100%'
                }
            },
            '& .MuiDialogActions-root': {
                bottom: 0
            }
        },
        paper: {
            minWidth: 240,
            overflow: 'hidden',
            minHeight: '100%',
            borderColor: base_tokens_json_1.default.tokens.palette.grey.grey100
        }
    }
};
exports.default = exports.MuiDrawer;
