// @oclc/dist/cjs/themes/components/fab.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiFab = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiFab = {
    defaultProps: {
        disableFocusRipple: true,
        disableRipple: true
    },
    styleOverrides: {
        root: {
            textTransform: 'none',
            minHeight: 'unset',
            width: 'unset',
            height: 'unset',
            '&:hover': {
                background: base_tokens_json_1.default.tokens.palette.blue.blue800
            },
            '&.Mui-focusVisible': {
                background: base_tokens_json_1.default.tokens.palette.blue.blue800,
                outline: `2px solid ${base_tokens_json_1.default.tokens.palette.text.text300}`,
                outlineOffset: '2px',
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            },
            '&:active': {
                background: base_tokens_json_1.default.tokens.palette.blue.blue700
            },
            '&.Mui-disabled': {
                background: base_tokens_json_1.default.tokens.palette.grey.disabled,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                cursor: 'not-allowed',
                pointerEvents: 'unset',
                '&:active': {
                    pointerEvents: 'none'
                }
            }
        },
        circular: ({ theme }) => ({
            '&.MuiFab-sizeSmall': {
                padding: theme.spacing(1),
                borderRadius: 20
            },
            '&.MuiFab-sizeMedium': {
                padding: theme.spacing(1),
                borderRadius: 24
            },
            '&.MuiFab-sizeLarge': {
                padding: theme.spacing(1.5),
                borderRadius: 28
            }
        }),
        extended: ({ theme }) => ({
            '&.MuiFab-sizeSmall': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 12,
                paddingRight: 12,
                borderRadius: 20,
                ...theme.typography.labelSm
            },
            '&.MuiFab-sizeMedium': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 12,
                paddingRight: 12,
                borderRadius: 24,
                ...theme.typography.label
            },
            '&.MuiFab-sizeLarge': {
                textTransform: 'none',
                paddingTop: theme.spacing(1.5),
                paddingBottom: theme.spacing(1.5),
                paddingLeft: 12,
                paddingRight: 12,
                borderRadius: 28,
                ...theme.typography.labelLg
            },
            '.MuiSvgIcon-root': {
                marginRight: theme.spacing(1)
            }
        })
    }
};
