// @oclc/dist/cjs/themes/components/fields.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiNativeSelect = exports.MuiSelect = exports.MuiAutocomplete = exports.MuiTextField = exports.MuiOutlinedInput = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const mdi_material_ui_1 = require("mdi-material-ui");
exports.MuiOutlinedInput = {
    defaultProps: {
        notched: false
    },
    styleOverrides: {
        root: ({ theme }) => ({
            '&.MuiOutlinedInput-root': {
                '&& .MuiAutocomplete-endAdornment': {
                    right: 8
                }
            },
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            boxSizing: 'border-box',
            '&:hover': {
                borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700
            },
            '&.Mui-focused': {
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: '0px 0px 0px 2px rgba(16, 116, 195, 0.59)',
                '& .MuiOutlinedInput-notchedOutline': {
                    border: `1px solid ${base_tokens_json_1.default.tokens.palette.blue.blue700}`
                },
                '& .MuiNativeSelect-select': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
                }
            },
            '&.Mui-error': {
                boxShadow: '0px 0px 0px 2px rgba(196, 32, 20, 0.5)',
                '& .MuiOutlinedInput-notchedOutline': {
                    border: `1px solid ${base_tokens_json_1.default.tokens.palette.red.red600}`
                }
            },
            '& .MuiSvgIcon-root': {
                color: base_tokens_json_1.default.tokens.palette.text.text100
            },
            '& .MuiSvgIcon-colorDisabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            },
            '& .MuiOutlinedInput-notchedOutline': {
                borderColor: base_tokens_json_1.default.tokens.palette.grey.grey600
            },
            '&:hover:not(.Mui-disabled)': {
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                ':not(.Mui-error) .MuiOutlinedInput-notchedOutline': {
                    borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                },
                '& .MuiSvgIcon-root': {
                    color: 'tokenData.tokens.palette.text.text300'
                }
            },
            '&.Mui-disabled': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '& .MuiOutlinedInput-notchedOutline': {
                    borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '& .MuiOutlinedInput-input': {
                    WebkitTextFillColor: base_tokens_json_1.default.tokens.palette.text.text300,
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '& .MuiNativeSelect-select': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 8,
                paddingRight: 8,
                color: base_tokens_json_1.default.tokens.palette.text.text100,
                backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
                '&:hover:not(.Mui-disabled)': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    ':not(.Mui-error) .MuiOutlinedInput-notchedOutline': {
                        borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                    }
                },
                '&.Mui-focused': {
                    '& .MuiSvgIcon-root': {
                        color: base_tokens_json_1.default.tokens.palette.text.text300
                    }
                }
            }
        }),
        input: {
            WebkitFontSmoothing: 'antialiased',
            MozOsxFontSmoothing: 'grayscale',
            '&.MuiInputBase-inputAdornedStart': {
                paddingLeft: '4px'
            },
            '&.MuiInputBase-inputMultiline': {
                paddingLeft: '0px'
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                WebkitTextFillColor: base_tokens_json_1.default.tokens.palette.text.text300,
                pointerEvents: 'unset',
                cursor: 'not-allowed'
            }
        },
        adornedStart: {
            paddingLeft: '4px'
        },
        adornedEnd: {
            paddingRight: '4px'
        },
        multiline: {
            padding: '8px'
        }
    }
};
exports.MuiTextField = {
    defaultProps: {
        size: 'small',
        InputLabelProps: { shrink: false },
        variant: 'outlined'
    },
    styleOverrides: {
        root: ({ theme }) => ({
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
            '& .MuiOutlinedInput-root': {
                ...theme.typography.body1,
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                boxSizing: 'border-box',
                '& > fieldset': {
                    border: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey400}`
                },
                '&.Mui-focused': {
                    borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
                    boxShadow: '0px 0px 0px 2px rgba(16, 116, 195, 0.59)',
                    '& > fieldset': {
                        border: `1px solid ${base_tokens_json_1.default.tokens.palette.blue.blue700}`
                    }
                },
                '&.Mui-disabled': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100,
                    '& > fieldset': {
                        borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled
                    }
                },
                '&.Mui-error': {
                    borderColor: base_tokens_json_1.default.tokens.palette.red.red600,
                    boxShadow: '0px 0px 0px 2px rgba(196, 32, 20, 0.5)',
                    '& > fieldset': {
                        border: `1px solid ${base_tokens_json_1.default.tokens.palette.red.red600}`
                    }
                },
                '&:hover:not(.Mui-disabled):not(.Mui-error)': {
                    '& > fieldset': {
                        borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                    }
                },
                '& .MuiOutlinedInput-input': {
                    '&::placeholder': {
                        color: base_tokens_json_1.default.tokens.palette.text.text100,
                        opacity: 1
                    },
                    '&.Mui-disabled': {
                        color: base_tokens_json_1.default.tokens.palette.text.text300,
                        WebkitTextFillColor: base_tokens_json_1.default.tokens.palette.text.text300
                    }
                },
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.text.text100
                },
                '& .MuiSvgIcon-colorDisabled': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '&.MuiInputBase-adornedStart': {
                    paddingLeft: '4px',
                    '& .MuiOutlinedInput-input': {
                        paddingLeft: '4px'
                    }
                },
                '&.MuiInputBase-adornedEnd .MuiInputAdornment-positionEnd': {
                    paddingRight: '4px',
                    marginLeft: 0,
                    '& .MuiOutlinedInput-input': {
                        paddingRight: '4px'
                    },
                    '& .MuiIconButton-root': {
                        marginRight: '-4px'
                    }
                },
                '& .MuiAutocomplete-input': {
                    paddingTop: theme.spacing(1),
                    paddingBottom: theme.spacing(1),
                    paddingLeft: 4,
                    paddingRight: 4
                }
            },
            '& .MuiOutlinedInput-input': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 8,
                paddingRight: 8,
                ...theme.typography.body1,
                height: 'auto'
            },
            '& .MuiInputBase-multiline': {
                padding: 0,
                '& .MuiOutlinedInput-input': {
                    paddingTop: theme.spacing(2),
                    paddingBottom: theme.spacing(2),
                    paddingLeft: 8,
                    paddingRight: 8
                }
            },
            '& .MuiSelect-select': {
                color: base_tokens_json_1.default.tokens.palette.text.text100,
                '&:hover:not(.Mui-disabled)': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    ':not(.Mui-error) .MuiOutlinedInput-notchedOutline': {
                        borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                    },
                    '& .MuiSvgIcon-root': {
                        color: base_tokens_json_1.default.tokens.palette.text.text300
                    }
                }
            },
            '& .MuiInputLabel-root': {
                marginBottom: '4px'
            }
        })
    }
};
exports.MuiAutocomplete = {
    defaultProps: {
        popupIcon: (0, jsx_runtime_1.jsx)(mdi_material_ui_1.ChevronDown, {}),
        size: 'small'
    },
    styleOverrides: {
        root: ({ theme }) => ({
            '& .MuiOutlinedInput-root.MuiInputBase-sizeSmall': {
                '&.MuiOutlinedInput-root.MuiInputBase-sizeSmall': {
                    paddingTop: theme.spacing(1),
                    paddingBottom: theme.spacing(1),
                    paddingLeft: 8,
                    paddingRight: 48,
                    '& .MuiAutocomplete-input': {
                        padding: 0,
                        ...theme.typography.body1
                    }
                }
            },
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
            '&.MuiOutlinedInput.MuiInputBase-sizeSmall': {
                paddingTop: theme.spacing(0.5),
                paddingBottom: theme.spacing(0.5),
                paddingLeft: 2,
                paddingRight: 2
            },
            '& .MuiOutlinedInput-notchedOutline': {
                borderRadius: 6
            },
            '& .MuiOutlinedInput-root': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 4,
                paddingRight: 4,
                '& .MuiAutocomplete-input': {
                    paddingTop: theme.spacing(1),
                    paddingBottom: theme.spacing(1),
                    paddingLeft: 4,
                    paddingRight: 4
                },
                '&.MuiInputBase-adornedStart': {
                    paddingLeft: 4
                },
                '&.MuiInputBase-adornedEnd': {
                    paddingRight: 4
                }
            },
            '& .MuiAutocomplete-input': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 4,
                paddingRight: 4,
                height: 'auto'
            },
            '& .MuiTextField-root': {
                '& .MuiOutlinedInput-input': {
                    paddingTop: theme.spacing(1),
                    paddingBottom: theme.spacing(1),
                    paddingLeft: 4,
                    paddingRight: 4
                },
                '& .MuiAutocomplete-input': {
                    minWidth: 80
                }
            },
            '& .MuiInputAdornment-positionStart': {
                '& .MuiOutlinedInput-root': {
                    borderTopRightRadius: 0,
                    borderBottomRightRadius: 0,
                    paddingRight: 0,
                    '&.MuiAutocomplete-hasPopupIcon': {
                        paddingRight: 100
                    },
                    '&.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon': {
                        paddingRight: 0
                    },
                    '& .MuiOutlinedInput-notchedOutline': {
                        borderColor: base_tokens_json_1.default.tokens.palette.grey.grey600,
                        borderTop: 'none',
                        borderBottom: 'none',
                        borderLeft: 'none',
                        borderTopRightRadius: 0,
                        borderBottomRightRadius: 0
                    }
                },
                '&.MuiInputAdornment-root': {
                    marginRight: 4
                },
                '& .MuiSelect-select': {
                    paddingRight: 32
                }
            },
            '& .MuiInputAdornment-positionEnd': {
                position: 'absolute',
                right: 0,
                height: '100%',
                '& .MuiCircularProgress-root': {
                    marginRight: 8
                },
                '& .MuiButton-contained': {
                    position: 'absolute',
                    top: 0,
                    right: 0,
                    height: '100%'
                },
                '& .MuiOutlinedInput-root': {
                    paddingRight: 40
                }
            },
            '& .MuiSelect-select': {
                paddingRight: 40,
                '& .MuiOutlinedInput-notchedOutline': {
                    border: 'none !important'
                }
            },
            '& .MuiAutocomplete-tag': {
                margin: 0,
                marginRight: 4
            }
        }),
        input: {
            padding: 0
        },
        inputRoot: {
            rowGap: 4
        },
        groupUl: {
            '& .MuiAutocomplete-option': {
                paddingLeft: 16
            }
        }
    }
};
exports.MuiSelect = {
    defaultProps: {
        IconComponent: mdi_material_ui_1.ChevronDown,
        size: 'small'
    },
    styleOverrides: {
        root: {
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
            '& .MuiOutlinedInput-notchedOutline': {
                borderColor: base_tokens_json_1.default.tokens.palette.grey.grey400
            },
            '&:hover:not(.Mui-disabled)': {
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                ':not(.Mui-error) .MuiOutlinedInput-notchedOutline': {
                    borderColor: base_tokens_json_1.default.tokens.palette.blue.blue700
                },
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '&.Mui-focused': {
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: '0px 0px 0px 2px rgba(16, 116, 195, 0.59)',
                '& .MuiOutlinedInput-notchedOutline': {
                    border: `1px solid ${base_tokens_json_1.default.tokens.palette.blue.blue700}`
                },
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '&.Mui-error': {
                borderColor: base_tokens_json_1.default.tokens.palette.red.red600,
                boxShadow: '0px 0px 0px 2px rgba(196, 32, 20, 0.5)',
                '& .MuiOutlinedInput-notchedOutline': {
                    border: `1px solid ${base_tokens_json_1.default.tokens.palette.red.red600}`
                }
            },
            '&.Mui-disabled': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100,
                '& .MuiOutlinedInput-notchedOutline': {
                    borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '& .MuiSvgIcon-root': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '& .MuiOutlinedInput-input': {
                    WebkitTextFillColor: base_tokens_json_1.default.tokens.palette.text.text300,
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            //TODO: Come back to this as it breaks autocompletes next to selects but also breaks the composite component when both are needed together
            // '& + .MuiAutocomplete-root': {
            //     '& .MuiOutlinedInput-root': {
            //         borderTopLeftRadius: 0,
            //         borderBottomLeftRadius: 0,
            //         zIndex: 1,
            //         '&:not(.Mui-disabled):not(.Mui-focused):not(.Mui-error)': {
            //             '& .MuiOutlinedInput-notchedOutline': {
            //                 borderLeft: 'none'
            //             }
            //         }
            //     }
            // },
            '&.MuiInputBase-adornedStart': {
                paddingLeft: 0,
                '& .MuiSelect-select': {
                    paddingLeft: 32
                },
                '& .MuiSvgIcon-root:first-of-type': {
                    pointerEvents: 'none',
                    position: 'absolute',
                    left: 8
                }
            }
        },
        icon: {
            color: base_tokens_json_1.default.tokens.palette.text.text100
        },
        select: ({ theme }) => ({
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 8,
            paddingRight: 8,
            minHeight: 'auto',
            ...theme.typography.body1,
            '&:hover:not(.Mui-disabled)': {
                color: base_tokens_json_1.default.tokens.palette.text.text300
            },
            '&.Mui-focused': {
                color: base_tokens_json_1.default.tokens.palette.text.text300
            }
        }),
        multiple: {
            color: base_tokens_json_1.default.tokens.palette.text.text100
        }
    }
};
exports.MuiNativeSelect = {
    defaultProps: {
        size: 'small',
        IconComponent: mdi_material_ui_1.ChevronDown
    },
    styleOverrides: {
        select: {
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.white
        },
        icon: {
            color: base_tokens_json_1.default.tokens.palette.text.text100
        }
    }
};
