// @oclc/dist/cjs/themes/components/labels.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiFormHelperText = exports.MuiFormControlLabel = exports.MuiFormLabel = exports.MuiInputLabel = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiInputLabel = {
    styleOverrides: {
        root: ({ theme }) => ({
            position: 'relative',
            ...theme.typography.label,
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            WebkitFontSmoothing: 'antialiased',
            MozOsxFontSmoothing: 'grayscale',
            transform: 'none',
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            },
            '&.Mui-focused': {
                color: base_tokens_json_1.default.tokens.palette.text.text300
            },
            '&.Mui-error': {
                color: base_tokens_json_1.default.tokens.palette.text.text300
            }
        })
    }
};
exports.MuiFormLabel = {
    styleOverrides: {
        root: ({ theme }) => ({
            ...theme.typography.label,
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            WebkitFontSmoothing: 'antialiased',
            MozOsxFontSmoothing: 'grayscale',
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            }
        })
    }
};
exports.MuiFormControlLabel = {
    styleOverrides: {
        label: {
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            }
        },
        root: {
            '& .MuiRadio-root + .MuiFormControlLabel-label': {
                marginLeft: '4px'
            },
            '& .MuiSwitch-root + .MuiFormControlLabel-label': {
                marginLeft: '8px'
            },
            '& .MuiCheckbox-root + .MuiFormControlLabel-label': {
                marginLeft: '8px'
            }
        }
    }
};
exports.MuiFormHelperText = {
    styleOverrides: {
        contained: ({ theme }) => ({
            marginLeft: '0px',
            marginRight: '0px',
            ...theme.typography.body2,
            color: base_tokens_json_1.default.tokens.palette.text.text100,
            '&.Mui-error': {
                color: base_tokens_json_1.default.tokens.palette.red.red600
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled
            }
        }),
        root: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.text.text100,
            ...theme.typography.body2,
            marginTop: 4,
            verticalAlign: 'middle',
            '& .MuiSvgIcon-root, svg, span': {
                verticalAlign: 'middle'
            },
            '& svg': {
                marginRight: '4px'
            }
        })
    }
};
