// @oclc/dist/cjs/themes/components/list.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiListSubheader = exports.MuiListItemText = exports.MuiListItemButton = exports.MuiListItem = exports.MuiList = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiList = {
    styleOverrides: {
        root: ({ theme }) => ({
            minWidth: '200px',
            'dt, dt.OclcDefinitionTerm': {
                fontWeight: 700
            },
            '& div': {
                '&:focus': {
                    outline: 'none',
                    '& .MuiMenuItem-root': {
                        background: base_tokens_json_1.default.tokens.palette.grey.grey100
                    }
                }
            },
            '& .MuiCollapse-root': {
                paddingLeft: '12px',
                paddingRight: '12px',
                '& > div > div > ul': {
                    padding: 0
                }
            },
            '& > div.MuiListItemButton-root': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 8,
                paddingRight: 8,
                '&.MuiListItemButton-dense': {
                    paddingTop: theme.spacing(0.25),
                    paddingBottom: theme.spacing(0.25),
                    paddingLeft: 8,
                    paddingRight: 8
                }
            },
            '&.MuiMultiSectionDigitalClockSection-root': {
                minWidth: 0
            }
        }),
        dense: ({ theme }) => ({
            '& .MuiMenuItem-root': {
                paddingTop: theme.spacing(0.25),
                paddingBottom: theme.spacing(0.25),
                paddingLeft: 8,
                paddingRight: 8,
                ...theme.typography.body1,
                minHeight: 'auto'
            }
        })
    }
};
exports.MuiListItem = {
    styleOverrides: {
        root: ({ theme }) => ({
            padding: 0,
            '& .MuiListItemButton-root': {
                paddingTop: theme.spacing(1),
                paddingBottom: theme.spacing(1),
                paddingLeft: 8,
                paddingRight: 8
            },
            '& .MuiListItemIcon-root': {
                color: base_tokens_json_1.default.tokens.palette.text.text100,
                minWidth: 32
            }
        }),
        dense: ({ theme }) => ({
            '& .MuiListItemButton-root': {
                paddingTop: theme.spacing(0.25),
                paddingBottom: theme.spacing(0.25),
                paddingLeft: 8,
                paddingRight: 8
            },
            '& .MuiSwitch-root': {
                marginBottom: 4
            },
            '& .MuiListItemIcon-root': {
                minWidth: 28
            },
            '& .MuiListItemText-inset': {
                paddingLeft: 28
            }
        })
    }
};
exports.MuiListItemButton = {
    styleOverrides: {
        root: {
            padding: 0,
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100
            },
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100,
                outline: '2px solid #000000 !important',
                outlineOffset: -2
            },
            '&.Mui-selected': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue200
            },
            '& .MuiListItemIcon-root': {
                color: base_tokens_json_1.default.tokens.palette.text.text100,
                minWidth: 28
            },
            '& .MuiListItemText-inset': {
                paddingLeft: 28
            }
        }
    }
};
exports.MuiListItemText = {
    styleOverrides: {
        root: {
            margin: 0
        },
        primary: {
            color: base_tokens_json_1.default.tokens.palette.text.text300
        },
        secondary: {
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            fontSize: '0.75rem'
        },
        inset: {
            paddingLeft: 32
        }
    }
};
exports.MuiListSubheader = {
    styleOverrides: {
        root: {
            color: base_tokens_json_1.default.tokens.palette.blue.blue700,
            padding: '0px 8px',
            lineHeight: '24px',
            fontWeight: 500,
            fontSize: '0.875rem'
        }
    }
};
