// @oclc/dist/cjs/themes/components/menu.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiMenuItem = exports.MuiMenu = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiMenu = {
    styleOverrides: {
        paper: {
            minWidth: '200px'
        }
    }
};
exports.MuiMenuItem = {
    styleOverrides: {
        root: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            whiteSpace: 'break-spaces',
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 8,
            paddingRight: 8,
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100,
                outline: '2px solid #000000',
                outlineOffset: -2
            },
            '&:hover': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100
            },
            '&. MuiListItemText-primary': {
                fontWeight: '700'
            },
            '&.Mui-selected': {
                fontWeight: '700',
                backgroundColor: 'unset',
                '& .MuiTypography-root': {
                    fontWeight: '700'
                },
                '&. MuiListItemText-root, .MuiTypography-root': {
                    fontWeight: '700',
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100
                },
                '&:hover, &:focus': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey100
                }
            },
            '& .MuiCheckbox-root': {
                alignSelf: 'flex-start'
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                opacity: 1
            },
            '& .MuiListItemIcon-root': {
                color: base_tokens_json_1.default.tokens.palette.text.text100,
                minWidth: 28
            }
        })
    }
};
