// @oclc/dist/cjs/themes/components/misc.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiBadge = exports.MuiAccordionDetails = exports.MuiAccordionSummary = exports.MuiAccordion = exports.MuiLinearProgress = exports.MuiCircularProgress = exports.MuiBreadcrumbs = exports.MuiPaginationItem = exports.MuiPagination = exports.MuiFormControl = exports.MuiAlert = exports.MuiLink = exports.MuiTooltip = exports.MuiDivider = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const Alert_1 = __importDefault(require("mdi-material-ui/Alert"));
const AlertBox_1 = __importDefault(require("mdi-material-ui/AlertBox"));
const CheckCircle_1 = __importDefault(require("mdi-material-ui/CheckCircle"));
const ChevronRight_1 = __importDefault(require("mdi-material-ui/ChevronRight"));
const InformationVariantCircle_1 = __importDefault(require("mdi-material-ui/InformationVariantCircle"));
const ChevronDown_1 = __importDefault(require("mdi-material-ui/ChevronDown"));
exports.MuiDivider = {
    styleOverrides: {
        root: {
            borderColor: base_tokens_json_1.default.tokens.palette.grey.grey500,
            '&:before, &:after': {
                borderColor: base_tokens_json_1.default.tokens.palette.grey.grey500
            }
        }
    }
};
exports.MuiTooltip = {
    defaultProps: {
        arrow: true
    },
    styleOverrides: {
        tooltip: ({ theme, ownerState }) => ({
            backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey600,
            color: base_tokens_json_1.default.tokens.palette.background.white,
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 6,
            paddingRight: 6,
            ...theme.typography.body1,
            boxShadow: '0px 1px 8px 0px rgba(0, 0, 0, 0.12), 0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 3px 3px -2px rgba(0, 0, 0, 0.2)',
            ...(!ownerState.arrow && {
                '&.MuiTooltip-tooltipPlacementBottom': {
                    marginTop: '7px !important',
                    marginBottom: '7px !important'
                },
                '&.MuiTooltip-tooltipPlacementTop': {
                    marginTop: '7px !important',
                    marginBottom: '7px !important'
                }
            })
        }),
        tooltipPlacementBottom: {
            '& .MuiTooltip-arrow': {
                filter: 'none'
            }
        },
        arrow: {
            color: base_tokens_json_1.default.tokens.palette.grey.grey600,
            filter: 'drop-shadow(2px 3px 6px black)'
        }
    }
};
exports.MuiLink = {
    styleOverrides: {
        root: {
            color: base_tokens_json_1.default.tokens.palette.blue.blue700,
            textDecorationColor: base_tokens_json_1.default.tokens.palette.blue.blue700,
            '&:hover': {
                color: base_tokens_json_1.default.tokens.palette.blue.blue800,
                textDecorationColor: base_tokens_json_1.default.tokens.palette.blue.blue800,
                textDecorationThickness: '2px'
            },
            '&:visited': {
                color: base_tokens_json_1.default.tokens.palette.purple.purple500,
                textDecorationColor: base_tokens_json_1.default.tokens.palette.purple.purple500
            },
            '&.Mui-focusVisible': {
                outlineOffset: '3px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                borderRadius: '6px',
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            }
        }
    }
};
exports.MuiAlert = {
    defaultProps: {
        variant: 'outlined',
        iconMapping: {
            error: (0, jsx_runtime_1.jsx)(AlertBox_1.default, { fontSize: "medium" }),
            warning: (0, jsx_runtime_1.jsx)(Alert_1.default, { fontSize: "medium" }),
            success: (0, jsx_runtime_1.jsx)(CheckCircle_1.default, { fontSize: "medium" }),
            info: (0, jsx_runtime_1.jsx)(InformationVariantCircle_1.default, { color: "info", fontSize: "medium" })
        }
    },
    styleOverrides: {
        root: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 6,
            paddingRight: 4,
            alignItems: 'center',
            boxShadow: '0px 2px 1px rgba(0, 0, 0, 0.2), 0px 2px 2px rgba(0, 0, 0, 0.14), 0px 1px 5px rgba(0, 0, 0, 0.12)',
            '& .MuiSvgIcon-root': {
                lineHeight: 1
            },
            '& .MuiAlertTitle-root': {
                marginTop: 0
            }
        }),
        outlined: {
            border: `1px solid transparent`,
            borderLeftWidth: '8px'
        },
        filled: {
            borderLeft: '8px solid transparent',
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.linen
        },
        standard: {
            boxShadow: 'none',
            backgroundColor: 'transparent'
        },
        outlinedSuccess: {
            borderColor: base_tokens_json_1.default.tokens.palette.green.green500,
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.green.green500
            }
        },
        filledSuccess: {
            borderLeftColor: base_tokens_json_1.default.tokens.palette.green.green500,
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.green.green500
            }
        },
        standardSuccess: {
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.green.green500
            }
        },
        outlinedInfo: {
            borderColor: base_tokens_json_1.default.tokens.palette.purple.purple600
        },
        filledInfo: {
            borderLeftColor: base_tokens_json_1.default.tokens.palette.purple.purple600
        },
        outlinedWarning: {
            borderColor: base_tokens_json_1.default.tokens.palette.yellow.yellow400,
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.yellow.yellow400
            }
        },
        filledWarning: {
            borderLeftColor: base_tokens_json_1.default.tokens.palette.yellow.yellow400,
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.yellow.yellow400
            }
        },
        standardWarning: {
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.yellow.yellow400
            }
        },
        outlinedError: {
            borderColor: base_tokens_json_1.default.tokens.palette.red.red600,
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.red.red600
            }
        },
        filledError: {
            borderLeftColor: base_tokens_json_1.default.tokens.palette.red.red600,
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.red.red600
            }
        },
        standardError: {
            '& .MuiAlert-icon': {
                color: base_tokens_json_1.default.tokens.palette.red.red600
            }
        },
        message: ({ theme }) => ({
            padding: 0,
            ...theme.typography.body1,
            '& > a': {
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                '&:hover': {
                    textDecorationThickness: '2px'
                }
            }
        }),
        icon: {
            marginRight: 8,
            padding: 0
        },
        action: {
            marginRight: 0,
            paddingTop: 0,
            alignItems: 'center'
        }
    }
};
exports.MuiFormControl = {
    styleOverrides: {
        root: ({ ownerState, theme }) => ({
            ...(ownerState.component === 'fieldset' && {
                '& .MuiFormLabel-root': {
                    color: base_tokens_json_1.default.tokens.palette.text.text100,
                    ...theme.typography.label,
                    '&.Mui-focused:not(.Mui-error)': {
                        color: base_tokens_json_1.default.tokens.palette.text.text100
                    },
                    '&.Mui-error': {
                        color: base_tokens_json_1.default.tokens.palette.red.red600
                    }
                },
                '& .MuiFormControlLabel-root': {
                    marginBottom: '2px'
                }
            }),
            '& .MuiFormGroup-root': {
                marginLeft: '9px'
            },
            '& .MuiFormGroup-row': {
                '& .MuiFormControlLabel-root': {
                    marginTop: '11px',
                    marginRight: '24px',
                    marginBottom: '10px'
                }
            },
            '&.MuiFormControl-marginDense': {
                margin: theme.spacing(1, 0, 0.5, 0)
            },
            '&.MuiFormControl-marginNormal': {
                margin: theme.spacing(2, 0, 1, 0)
            }
        })
    }
};
exports.MuiPagination = {
    defaultProps: {
        showFirstButton: true,
        showLastButton: true,
        shape: 'rounded',
        boundaryCount: 0
    }
};
exports.MuiPaginationItem = {
    styleOverrides: {
        root: ({ theme }) => ({
            minHeight: 'unset',
            height: theme.spacing(6.5),
            minWidth: theme.spacing(6.5),
            ...theme.typography.body1,
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            margin: '0 2px',
            '&.Mui-selected': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue200,
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                fontWeight: 600,
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue600,
                    color: base_tokens_json_1.default.tokens.palette.background.white
                }
            },
            '&.MuiPaginationItem-ellipsis': {
                display: 'none'
            },
            '&:hover:not(.MuiPaginationItem-ellipsis)': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue600,
                color: base_tokens_json_1.default.tokens.palette.background.white
            },
            '&.Mui-disabled': {
                opacity: 1,
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                '& .MuiButton-root': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    '& .MuiButton-startIcon': {
                        color: base_tokens_json_1.default.tokens.palette.grey.disabled
                    },
                    '& .MuiButton-endIcon': {
                        color: base_tokens_json_1.default.tokens.palette.grey.disabled
                    }
                },
                '&.Mui-selected': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled
                }
            },
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue600,
                color: base_tokens_json_1.default.tokens.palette.background.white,
                outlineOffset: '3px',
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                '&.Mui-selected': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue600,
                    color: base_tokens_json_1.default.tokens.palette.background.white,
                    outlineOffset: '3px',
                    outlineStyle: 'solid',
                    outlineWidth: '2px',
                    outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                    boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                }
            },
            '& .MuiButton-root': {
                fontSize: base_tokens_json_1.default.tokens.typography.body.fontSize,
                fontWeight: 400,
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    backgroundColor: 'transparent'
                },
                '& .MuiButton-startIcon': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                },
                '& .MuiButton-endIcon': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            },
            '& .MuiTouchRipple-root': {
                display: 'none'
            }
        }),
        outlined: {
            borderColor: base_tokens_json_1.default.tokens.palette.grey.grey300,
            '&.Mui-selected': {
                borderColor: base_tokens_json_1.default.tokens.palette.grey.grey500,
                backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue200,
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                '&:hover': {
                    borderColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue500,
                    backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue600,
                    color: base_tokens_json_1.default.tokens.palette.background.white
                }
            },
            '&:hover': {
                borderColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue500,
                backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
                color: base_tokens_json_1.default.tokens.palette.blue.blue700
            },
            '&.Mui-disabled': {
                opacity: 1,
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                '& .MuiButton-root': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    borderColor: base_tokens_json_1.default.tokens.palette.grey.disabled,
                    '& .MuiButton-startIcon': {
                        color: base_tokens_json_1.default.tokens.palette.grey.disabled
                    },
                    '& .MuiButton-endIcon': {
                        color: base_tokens_json_1.default.tokens.palette.grey.disabled
                    }
                }
            },
            '& .MuiButton-root': {
                fontSize: base_tokens_json_1.default.tokens.typography.body.fontSize,
                fontWeight: 400,
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.blue.blue700,
                    backgroundColor: 'transparent',
                    '& .MuiButton-startIcon': {
                        color: base_tokens_json_1.default.tokens.palette.blue.blue700
                    },
                    '& .MuiButton-endIcon': {
                        color: base_tokens_json_1.default.tokens.palette.blue.blue700
                    }
                },
                '& .MuiButton-startIcon': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                },
                '& .MuiButton-endIcon': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                }
            }
        },
        sizeSmall: ({ theme }) => ({
            height: theme.spacing(6),
            minWidth: theme.spacing(6),
            ...(theme.spacing(1) === '4px' ? theme.typography.body1 : theme.typography.body2),
            margin: '0 2px'
        }),
        sizeLarge: ({ theme }) => ({
            height: theme.spacing(8),
            minWidth: theme.spacing(8),
            ...theme.typography.body1
        })
    }
};
exports.MuiBreadcrumbs = {
    defaultProps: {
        separator: (0, jsx_runtime_1.jsx)(ChevronRight_1.default, {})
    },
    styleOverrides: {
        root: ({ theme }) => ({
            ...theme.typography.body2,
            color: base_tokens_json_1.default.tokens.palette.text.text100
        }),
        separator: {
            marginLeft: 4,
            marginRight: 4,
            '& .MuiSvgIcon-root': {
                fontSize: 16,
                color: base_tokens_json_1.default.tokens.palette.grey.grey400
            }
        },
        ol: {
            '& > li > button': {
                backgroundColor: 'transparent',
                color: '#000000',
                '&:hover': {
                    backgroundColor: base_tokens_json_1.default.tokens.palette.purple.purple100
                },
                '&.Mui-focusVisible': {
                    textDecoration: 'none',
                    borderRadius: '6px',
                    outlineOffset: '2px',
                    outlineStyle: 'solid',
                    outlineWidth: '2px',
                    outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                    boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey200,
                    color: base_tokens_json_1.default.tokens.palette.text.text300
                },
                '& .MuiTouchRipple-root': {
                    display: 'none'
                }
            }
        },
        li: ({ theme }) => ({
            '& .MuiLink-root': {
                display: 'flex',
                alignItems: 'center',
                columnGap: '4px',
                ...theme.typography.body2,
                color: base_tokens_json_1.default.tokens.palette.text.text100,
                '&:visited': {
                    color: base_tokens_json_1.default.tokens.palette.text.text100,
                    textDecorationColor: base_tokens_json_1.default.tokens.palette.text.text100
                },
                '&:hover': {
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    textDecorationColor: base_tokens_json_1.default.tokens.palette.text.text300,
                    textDecorationThickness: '2px'
                },
                '&.Mui-focusVisible': {
                    textDecoration: 'none',
                    borderRadius: '6px',
                    outlineOffset: '2px',
                    outlineStyle: 'solid',
                    outlineWidth: '2px',
                    outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                    backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey200,
                    color: base_tokens_json_1.default.tokens.palette.text.text300,
                    boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
                },
                '& .MuiSvgIcon-root': {
                    fontSize: '16px'
                }
            },
            '& > p': {
                ...theme.typography.body2,
                color: base_tokens_json_1.default.tokens.palette.text.text300,
                display: 'flex',
                alignItems: 'center',
                columnGap: '4px',
                '& .MuiSvgIcon-root': {
                    fontSize: '16px'
                }
            }
        })
    }
};
exports.MuiCircularProgress = {
    defaultProps: {
        thickness: 5.5
    },
    styleOverrides: {
        root: {
            '@media (prefers-reduced-motion)': {
                animationDuration: '3.4s'
            }
        },
        circle: {
            strokeLinecap: 'round'
        }
    }
};
exports.MuiLinearProgress = {
    styleOverrides: {
        root: {
            height: '8px',
            borderRadius: '17px',
            backgroundColor: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue200
        },
        bar: {
            borderRadius: '19px'
        }
    }
};
exports.MuiAccordion = {
    defaultProps: {
        square: true,
        disableGutters: true
    },
    styleOverrides: {
        root: {
            boxShadow: 'none',
            borderBottom: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey200}`,
            '&:before': {
                content: 'none'
            }
        }
    }
};
exports.MuiAccordionSummary = {
    defaultProps: {
        expandIcon: (0, jsx_runtime_1.jsx)(ChevronDown_1.default, { fontSize: "large" })
    },
    styleOverrides: {
        root: ({ theme }) => ({
            flexDirection: 'row-reverse',
            paddingTop: theme.spacing(2),
            paddingBottom: theme.spacing(2),
            paddingLeft: 12,
            paddingRight: 12,
            columnGap: '8px',
            minHeight: 'unset',
            ...theme.typography.label,
            '&.Mui-focusVisible': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100,
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                outlineOffset: '2px',
                zIndex: 999,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            },
            '&:hover:not(.Mui-disabled)': {
                backgroundColor: base_tokens_json_1.default.tokens.palette.blue.blue100
            },
            '&.Mui-disabled': {
                opacity: 1,
                '& .MuiAccordionSummary-content': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled
                },
                '& .MuiAccordionSummary-expandIconWrapper': {
                    color: base_tokens_json_1.default.tokens.palette.grey.disabled
                }
            }
        }),
        content: ({ theme }) => ({
            fontWeight: 600,
            color: base_tokens_json_1.default.tokens.palette.text.text300,
            ...theme.typography.label,
            margin: 0,
            '&.Mui-expanded': {
                margin: 0
            }
        }),
        expandIconWrapper: {
            color: base_tokens_json_1.default.tokens.palette.text.text100
        }
    }
};
exports.MuiAccordionDetails = {
    styleOverrides: {
        root: ({ theme }) => ({
            paddingTop: theme.spacing(3),
            paddingBottom: theme.spacing(3),
            paddingRight: 12,
            paddingLeft: 36,
            color: base_tokens_json_1.default.tokens.palette.text.text300
        })
    }
};
exports.MuiBadge = {
    defaultProps: {
        max: 99
    },
    styleOverrides: {
        badge: ({ theme }) => ({
            color: base_tokens_json_1.default.tokens.palette.background.white,
            backgroundColor: base_tokens_json_1.default.tokens.palette.purple.purple600,
            outline: `1px solid ${base_tokens_json_1.default.tokens.palette.background.white}`,
            ...theme.typography.labelSm,
            padding: '0px 5px',
            borderRadius: '100px'
        }),
        dot: {
            color: base_tokens_json_1.default.tokens.palette.background.white,
            backgroundColor: base_tokens_json_1.default.tokens.palette.purple.purple600,
            minHeight: '10px',
            minWidth: '10px',
            borderRadius: '100px'
        }
    }
};
