// @oclc/dist/cjs/themes/components/palette.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const palette = {
    primary: {
        light: base_tokens_json_1.default.tokens.palette.blue.blue300,
        main: base_tokens_json_1.default.tokens.palette.blue.blue700,
        dark: base_tokens_json_1.default.tokens.palette.blue.blue800
    },
    secondary: {
        light: base_tokens_json_1.default.tokens.palette.blue.blue600,
        main: base_tokens_json_1.default.tokens.palette.blue.blue700,
        dark: base_tokens_json_1.default.tokens.palette.blue.blue700
    },
    error: {
        light: base_tokens_json_1.default.tokens.palette.red.red300,
        main: base_tokens_json_1.default.tokens.palette.red.red600,
        dark: base_tokens_json_1.default.tokens.palette.red.red700
    },
    success: {
        light: base_tokens_json_1.default.tokens.palette.green.green400,
        main: base_tokens_json_1.default.tokens.palette.green.green500,
        dark: base_tokens_json_1.default.tokens.palette.green.green600
    },
    warning: {
        light: base_tokens_json_1.default.tokens.palette.yellow.yellow300,
        main: base_tokens_json_1.default.tokens.palette.yellow.yellow400,
        dark: base_tokens_json_1.default.tokens.palette.yellow.yellow600
    },
    info: {
        light: base_tokens_json_1.default.tokens.palette.purple.purple500,
        main: base_tokens_json_1.default.tokens.palette.purple.purple600,
        dark: base_tokens_json_1.default.tokens.palette.purple.purple700
    },
    text: {
        primary: base_tokens_json_1.default.tokens.palette.text.text300,
        secondary: base_tokens_json_1.default.tokens.palette.text.text200,
        disabled: base_tokens_json_1.default.tokens.palette.text.text100
    }
};
exports.default = palette;
