// @oclc/dist/cjs/themes/components/radio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const MuiRadio = {
    defaultProps: {
        disableRipple: true
    },
    styleOverrides: {
        root: {
            paddingLeft: 3,
            paddingRight: 3,
            padding: 3,
            color: base_tokens_json_1.default.tokens.palette.grey.grey500,
            '&:hover': {
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: base_tokens_json_1.default.tokens.palette.blue.blue700
            },
            '&.Mui-focusVisible': {
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineOffset: '1px',
                outlineColor: base_tokens_json_1.default.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${base_tokens_json_1.default.tokens.palette.background.white}`
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    outline: 'none',
                    backgroundColor: 'transparent'
                }
            }
        }
    }
};
exports.default = MuiRadio;
