// @oclc/dist/cjs/themes/components/snackbar.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiSnackbarContent = exports.MuiSnackbar = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
const Slide_1 = __importDefault(require("@mui/material/Slide"));
function SlideTransition(props) {
    return (0, jsx_runtime_1.jsx)(Slide_1.default, { ...props, direction: "right" });
}
exports.MuiSnackbar = {
    defaultProps: {
        TransitionComponent: SlideTransition
    },
    styleOverrides: {
        root: {
            maxWidth: '344px'
        }
    }
};
exports.MuiSnackbarContent = {
    styleOverrides: {
        root: ({ theme }) => ({
            backgroundColor: base_tokens_json_1.default.tokens.palette.grey.grey700,
            borderRadius: '4px',
            minHeight: 36,
            paddingTop: 10,
            paddingBottom: 10,
            paddingLeft: 16,
            paddingRight: 12,
            ...theme.typography.body1
        }),
        message: {
            color: base_tokens_json_1.default.tokens.palette.background.white,
            padding: 0
        },
        action: {
            '& .MuiButton-root': {
                color: base_tokens_json_1.default.tokens.palette.blue.blue300,
                fontSize: '0.875rem'
            },
            '& .MuiIconButton-root': {
                color: base_tokens_json_1.default.tokens.palette.background.white
            }
        }
    }
};
