// @oclc/dist/cjs/themes/components/table.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiTablePagination = exports.MuiTableBody = exports.MuiTableSortLabel = exports.MuiTableCell = exports.MuiTableHead = exports.MuiTableContainer = exports.MuiTable = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiTable = {
    styleOverrides: {
        root: {
            '&.OclcTable-striped': {
                '& .MuiTableRow-root': {
                    '&:nth-of-type(even)': {
                        background: base_tokens_json_1.default.tokens.palette.background.snow,
                        '&.Mui-selected': {
                            background: base_tokens_json_1.default.tokens.palette.background.snow
                        }
                    }
                },
                '& .MuiTableBody-root': {
                    '& .MuiTableRow-root': {
                        '&:nth-of-type(odd)': {
                            background: base_tokens_json_1.default.tokens.palette.background.white,
                            '&.Mui-selected': {
                                background: base_tokens_json_1.default.tokens.palette.background.white
                            }
                        }
                    }
                },
                '& .OclcTableBody-collapsible': {
                    '&:nth-of-type(even)': {
                        background: base_tokens_json_1.default.tokens.palette.background.snow,
                        '& .MuiTableRow-root': {
                            background: base_tokens_json_1.default.tokens.palette.background.snow
                        }
                    },
                    '&:nth-of-type(odd)': {
                        background: base_tokens_json_1.default.tokens.palette.background.white,
                        '& .MuiTableRow-root': {
                            background: base_tokens_json_1.default.tokens.palette.background.white
                        }
                    }
                }
            }
        }
    }
};
exports.MuiTableContainer = {
    styleOverrides: {
        root: {
            border: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey300}`,
            boxShadow: 'none'
        }
    }
};
exports.MuiTableHead = {
    styleOverrides: {
        root: {
            background: base_tokens_json_1.default.tokens.palette.darkBlue.darkBlue100,
            '& th': {
                '&:has(.Mui-focusVisible)': {
                    background: base_tokens_json_1.default.tokens.palette.background.white,
                    outline: '2px solid #000',
                    outlineOffset: '-2px'
                }
            },
            '& .MuiTableCell-paddingCheckbox': {
                textAlign: 'center',
            }
        }
    }
};
exports.MuiTableCell = {
    defaultProps: {
        size: 'small'
    },
    styleOverrides: {
        root: ({ theme }) => ({
            fontWeight: 500,
            borderBottom: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey300}`,
            ...theme.typography.body1,
            '&.MuiTableCell-head': {
                ...theme.typography.labelLg,
                paddingTop: theme.spacing(1.5),
                paddingBottom: theme.spacing(1.5),
                paddingLeft: 12,
                paddingRight: 8
            },
            '&.MuiTablePagination-root': {
                fontSize: 14,
                borderBottom: 'none',
                padding: 0
            }
        }),
        sizeSmall: ({ theme }) => ({
            ...theme.typography.body1,
            paddingTop: theme.spacing(1),
            paddingBottom: theme.spacing(1),
            paddingLeft: 12,
            paddingRight: 12
        })
    }
};
exports.MuiTableSortLabel = {
    styleOverrides: {
        root: {
            '&.MuiButtonBase-root': {
                color: `${base_tokens_json_1.default.tokens.palette.text.text300} !important`
            },
            '&:hover': {
                '& .MuiTableSortLabel-icon': {
                    opacity: 1
                }
            }
        }
    }
};
exports.MuiTableBody = {
    styleOverrides: {
        root: {
            position: 'relative',
            '&.OclcTableBody-collapsible': {
                '&:not(:last-child)': {
                    borderBottom: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey300}`
                },
                '& .MuiTableCell-root': {
                    borderBottom: 'none'
                },
                '&:has(.MuiCollapse-entered)': {
                    '&:before': {
                        content: '""',
                        position: 'absolute',
                        top: 0,
                        left: 0,
                        borderLeft: `2px solid ${base_tokens_json_1.default.tokens.palette.blue.blue700}`,
                        height: '100%'
                    }
                }
            },
            '& .MuiTableCell-paddingCheckbox': {
                borderBottom: 'none'
            }
        }
    }
};
exports.MuiTablePagination = {
    defaultProps: {
        labelRowsPerPage: 'Page size'
    },
    styleOverrides: {
        root: {
            '& .MuiTablePagination-input': {
                marginRight: '24px'
            }
        },
        selectLabel: ({ theme }) => ({
            ...theme.typography.body1
        }),
        select: {
            fontSize: '0.875rem',
            color: base_tokens_json_1.default.tokens.palette.blue.blue700,
            fontWeight: '600',
            marginLeft: 0
        },
        actions: {
            marginLeft: 8
        },
        displayedRows: ({ theme }) => ({
            ...theme.typography.body1
        })
    }
};
