// @oclc/dist/cjs/themes/components/tabs.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuiTab = exports.MuiTabs = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.MuiTabs = {
    styleOverrides: {
        root: {
            overflow: 'visible',
            alignItems: 'end'
        },
        fixed: {
            overflow: 'visible !important'
        },
        indicator: {
            display: 'none'
        },
        scroller: {
            padding: 4,
            paddingBottom: 0,
            paddingLeft: 0
        },
        flexContainer: {
            columnGap: '1.5px'
        },
        scrollButtons: ({ theme }) => ({
            border: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey200}`,
            padding: theme.spacing(1, 2),
            color: base_tokens_json_1.default.tokens.palette.text.text100,
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.snow,
            maxWidth: 28,
            '& .MuiSvgIcon-root': {
                fontSize: 18
            },
            '&:hover': {
                backgroundColor: 'rgba(16, 116, 195, 0.04)'
            },
            '&.Mui-disabled': {
                display: 'none'
            }
        })
    }
};
exports.MuiTab = {
    defaultProps: {
        disableRipple: true
    },
    styleOverrides: {
        root: ({ theme }) => ({
            textTransform: 'none',
            ...theme.typography.body1,
            minWidth: 0,
            color: base_tokens_json_1.default.tokens.palette.text.text200,
            backgroundColor: base_tokens_json_1.default.tokens.palette.background.snow,
            border: `1px solid ${base_tokens_json_1.default.tokens.palette.grey.grey200}`,
            borderBottom: 'none',
            padding: theme.spacing(1, 2),
            minHeight: 'unset',
            '&.Mui-selected': {
                boxShadow: `0px -4px 0px ${base_tokens_json_1.default.tokens.palette.blue.blue800}, 0 4px 0 -1px ${base_tokens_json_1.default.tokens.palette.background.white}`,
                color: base_tokens_json_1.default.tokens.palette.blue.blue700,
                backgroundColor: base_tokens_json_1.default.tokens.palette.background.white,
                borderBottom: 'none',
                ...theme.typography.label
            },
            '&:hover': {
                '&:not(.Mui-selected)': {
                    backgroundColor: 'rgba(16, 116, 195, 0.04)'
                }
            },
            '&.Mui-focusVisible': {
                '&:not(.Mui-selected)': {
                    backgroundColor: 'rgba(16, 116, 195, 0.04)'
                },
                outline: `2px solid ${base_tokens_json_1.default.tokens.palette.text.text300}`,
                outlineOffset: '-4px'
            },
            '&.Mui-disabled': {
                color: base_tokens_json_1.default.tokens.palette.grey.disabled,
                backgroundColor: base_tokens_json_1.default.tokens.palette.background.snow
            }
        })
    }
};
