// @oclc/dist/cjs/themes/components/typography.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.typographySmall = exports.typography = void 0;
const base_tokens_json_1 = __importDefault(require("../../design-tokens/base-tokens.json"));
exports.typography = {
    fontFamily: base_tokens_json_1.default.tokens.typography.fontFamily,
    fontSize: 16,
    h1: base_tokens_json_1.default.tokens.typography.heading3xl,
    h2: base_tokens_json_1.default.tokens.typography.heading2xl,
    h3: base_tokens_json_1.default.tokens.typography.headingXl,
    h4: base_tokens_json_1.default.tokens.typography.headingLg,
    h5: base_tokens_json_1.default.tokens.typography.headingMd,
    h6: base_tokens_json_1.default.tokens.typography.heading,
    body1: base_tokens_json_1.default.tokens.typography.body,
    body2: base_tokens_json_1.default.tokens.typography.bodySm,
    caption: base_tokens_json_1.default.tokens.typography.caption,
    button: base_tokens_json_1.default.tokens.typography.label,
    label: base_tokens_json_1.default.tokens.typography.label,
    labelSm: base_tokens_json_1.default.tokens.typography.labelSm,
    labelLg: base_tokens_json_1.default.tokens.typography.labelLg,
    subtitle1: base_tokens_json_1.default.tokens.typography.body,
    subtitle2: base_tokens_json_1.default.tokens.typography.bodySm,
    iconSm: base_tokens_json_1.default.tokens.typography.iconSm,
    icon: base_tokens_json_1.default.tokens.typography.icon,
    iconLg: base_tokens_json_1.default.tokens.typography.iconLg,
    chip: base_tokens_json_1.default.tokens.typography.chip
};
exports.typographySmall = {
    fontFamily: base_tokens_json_1.default.tokens.typography.compact.fontFamily,
    fontSize: 16,
    h1: base_tokens_json_1.default.tokens.typography.compact.heading3xl,
    h2: base_tokens_json_1.default.tokens.typography.compact.heading2xl,
    h3: base_tokens_json_1.default.tokens.typography.compact.headingXl,
    h4: base_tokens_json_1.default.tokens.typography.compact.headingLg,
    h5: base_tokens_json_1.default.tokens.typography.compact.headingMd,
    h6: base_tokens_json_1.default.tokens.typography.compact.heading,
    body1: base_tokens_json_1.default.tokens.typography.compact.body,
    body2: base_tokens_json_1.default.tokens.typography.compact.bodySm,
    caption: base_tokens_json_1.default.tokens.typography.compact.caption,
    button: base_tokens_json_1.default.tokens.typography.compact.label,
    label: base_tokens_json_1.default.tokens.typography.compact.label,
    labelSm: base_tokens_json_1.default.tokens.typography.compact.labelSm,
    labelLg: base_tokens_json_1.default.tokens.typography.compact.labelLg,
    subtitle1: base_tokens_json_1.default.tokens.typography.compact.body,
    subtitle2: base_tokens_json_1.default.tokens.typography.compact.bodySm,
    iconSm: base_tokens_json_1.default.tokens.typography.compact.iconSm,
    icon: base_tokens_json_1.default.tokens.typography.compact.icon,
    iconLg: base_tokens_json_1.default.tokens.typography.compact.iconLg,
    chip: base_tokens_json_1.default.tokens.typography.compact.chip
};
exports.default = exports.typography;
