// @oclc/dist/cjs/themes/basic-theme-v6.js
"use strict";
'use client';
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.basicThemeV6 = void 0;
const styles_1 = require("@mui/material/styles");
const base_tokens_json_1 = __importDefault(require("../design-tokens/base-tokens.json"));
const typography_1 = __importDefault(require("./components/typography"));
const spacing_1 = require("./components/spacing");
const palette_1 = __importDefault(require("./components/palette"));
const buttons_1 = require("./components/buttons");
const checkbox_1 = __importDefault(require("./components/checkbox"));
const radio_1 = __importDefault(require("./components/radio"));
const switch_1 = __importDefault(require("./components/switch"));
const icons_1 = require("./components/icons");
const fields_1 = require("./components/fields");
const chip_1 = __importDefault(require("./components/chip"));
const labels_1 = require("./components/labels");
const datepicker_v6_1 = require("./components/datepicker-v6");
const tabs_1 = require("./components/tabs");
const dialogs_1 = require("./components/dialogs");
const menu_1 = require("./components/menu");
const list_1 = require("./components/list");
const datagrid_1 = __importDefault(require("./components/datagrid"));
const table_1 = require("./components/table");
const misc_1 = require("./components/misc");
const stepper_1 = require("./components/stepper");
const snackbar_1 = require("./components/snackbar");
const togglebutton_1 = require("./components/togglebutton");
const card_1 = require("./components/card");
const avatar_1 = require("./components/avatar");
const appbar_1 = require("./components/appbar");
const drawer_1 = require("./components/drawer");
const slider_1 = require("./components/slider");
const fab_1 = require("./components/fab");
let theme = (0, styles_1.createTheme)({
    typography: typography_1.default,
    palette: palette_1.default,
    shadows: [
        'none',
        base_tokens_json_1.default.tokens.elevations[1],
        base_tokens_json_1.default.tokens.elevations[2],
        base_tokens_json_1.default.tokens.elevations[3],
        base_tokens_json_1.default.tokens.elevations[4],
        base_tokens_json_1.default.tokens.elevations[5],
        base_tokens_json_1.default.tokens.elevations[6],
        base_tokens_json_1.default.tokens.elevations[7],
        base_tokens_json_1.default.tokens.elevations[8],
        base_tokens_json_1.default.tokens.elevations[9],
        base_tokens_json_1.default.tokens.elevations[10],
        base_tokens_json_1.default.tokens.elevations[11],
        base_tokens_json_1.default.tokens.elevations[12],
        base_tokens_json_1.default.tokens.elevations[13],
        base_tokens_json_1.default.tokens.elevations[14],
        base_tokens_json_1.default.tokens.elevations[15],
        base_tokens_json_1.default.tokens.elevations[16],
        base_tokens_json_1.default.tokens.elevations[17],
        base_tokens_json_1.default.tokens.elevations[18],
        base_tokens_json_1.default.tokens.elevations[19],
        base_tokens_json_1.default.tokens.elevations[20],
        base_tokens_json_1.default.tokens.elevations[21],
        base_tokens_json_1.default.tokens.elevations[22],
        base_tokens_json_1.default.tokens.elevations[23],
        base_tokens_json_1.default.tokens.elevations[24]
    ],
    spacing: spacing_1.spacing
});
exports.basicThemeV6 = (0, styles_1.createTheme)(theme, {
    shape: {
        borderRadius: 6
    },
    components: {
        MuiButton: buttons_1.MuiButton,
        MuiIconButton: buttons_1.MuiIconButton,
        MuiButtonGroup: buttons_1.MuiButtonGroup,
        MuiCheckbox: checkbox_1.default,
        MuiRadio: radio_1.default,
        MuiSwitch: switch_1.default,
        MuiSvgIcon: icons_1.MuiSvgIcon,
        MuiIcon: icons_1.MuiIcon,
        MuiDivider: misc_1.MuiDivider,
        MuiTooltip: misc_1.MuiTooltip,
        MuiChip: chip_1.default,
        MuiLink: misc_1.MuiLink,
        MuiAlert: misc_1.MuiAlert,
        MuiTabs: tabs_1.MuiTabs,
        MuiTab: tabs_1.MuiTab,
        MuiBackdrop: dialogs_1.MuiBackdrop,
        MuiDataGrid: datagrid_1.default,
        MuiTable: table_1.MuiTable,
        MuiTableContainer: table_1.MuiTableContainer,
        MuiTableHead: table_1.MuiTableHead,
        MuiTableCell: table_1.MuiTableCell,
        MuiOutlinedInput: fields_1.MuiOutlinedInput,
        MuiTextField: fields_1.MuiTextField,
        MuiAutocomplete: fields_1.MuiAutocomplete,
        MuiSelect: fields_1.MuiSelect,
        MuiNativeSelect: fields_1.MuiNativeSelect,
        MuiTablePagination: table_1.MuiTablePagination,
        MuiTableSortLabel: table_1.MuiTableSortLabel,
        MuiTableBody: table_1.MuiTableBody,
        MuiModal: dialogs_1.MuiModal,
        MuiDialog: dialogs_1.MuiDialog,
        MuiDialogActions: dialogs_1.MuiDialogActions,
        MuiInputLabel: labels_1.MuiInputLabel,
        MuiFormLabel: labels_1.MuiFormLabel,
        MuiFormControlLabel: labels_1.MuiFormControlLabel,
        MuiFormHelperText: labels_1.MuiFormHelperText,
        MuiList: list_1.MuiList,
        MuiListItem: list_1.MuiListItem,
        MuiListItemButton: list_1.MuiListItemButton,
        MuiListItemText: list_1.MuiListItemText,
        MuiMenu: menu_1.MuiMenu,
        MuiMenuItem: menu_1.MuiMenuItem,
        MuiListSubheader: list_1.MuiListSubheader,
        MuiFormControl: misc_1.MuiFormControl,
        MuiPagination: misc_1.MuiPagination,
        MuiPaginationItem: misc_1.MuiPaginationItem,
        MuiAccordion: misc_1.MuiAccordion,
        MuiAccordionDetails: misc_1.MuiAccordionDetails,
        MuiAccordionSummary: misc_1.MuiAccordionSummary,
        MuiBadge: misc_1.MuiBadge,
        MuiBreadcrumbs: misc_1.MuiBreadcrumbs,
        MuiCircularProgress: misc_1.MuiCircularProgress,
        MuiLinearProgress: misc_1.MuiLinearProgress,
        MuiSnackbar: snackbar_1.MuiSnackbar,
        MuiSnackbarContent: snackbar_1.MuiSnackbarContent,
        MuiDateTimePicker: datepicker_v6_1.MuiDateTimePicker,
        MuiDatePicker: datepicker_v6_1.MuiDatePicker,
        MuiDatePickerToolbar: datepicker_v6_1.MuiDatePickerToolbar,
        MuiPickersDay: datepicker_v6_1.MuiPickersDay,
        MuiPickersCalendarHeader: datepicker_v6_1.MuiPickersCalendarHeader,
        MuiYearPicker: datepicker_v6_1.MuiYearPicker,
        MuiYearCalendar: datepicker_v6_1.MuiYearCalendar,
        MuiPickersYear: datepicker_v6_1.MuiPickersYear,
        MuiPickersArrowSwitcher: datepicker_v6_1.MuiPickersArrowSwitcher,
        MuiMultiInputDateRangeField: datepicker_v6_1.MuiMultiInputDateRangeField,
        MuiDateRangePickerDay: datepicker_v6_1.MuiDateRangePickerDay,
        // v6 additional styling
        MuiPickersMonth: datepicker_v6_1.MuiPickersMonth,
        MuiDayCalendar: datepicker_v6_1.MuiDayCalendar,
        MuiPickersLayout: datepicker_v6_1.MuiPickersLayout,
        MuiPickersToolbar: datepicker_v6_1.MuiPickersToolbar,
        MuiMultiSectionDigitalClock: datepicker_v6_1.MuiMultiSectionDigitalClock,
        MuiStepper: stepper_1.MuiStepper,
        MuiMobileStepper: stepper_1.MuiMobileStepper,
        MuiStepButton: stepper_1.MuiStepButton,
        MuiStep: stepper_1.MuiStep,
        MuiToggleButton: togglebutton_1.MuiToggleButton,
        MuiToggleButtonGroup: togglebutton_1.MuiToggleButtonGroup,
        MuiCard: card_1.MuiCard,
        MuiCardActions: card_1.MuiCardActions,
        MuiCardMedia: card_1.MuiCardMedia,
        MuiCardHeader: card_1.MuiCardHeader,
        MuiCardActionArea: card_1.MuiCardActionArea,
        MuiCardContent: card_1.MuiCardContent,
        MuiAvatar: avatar_1.MuiAvatar,
        MuiAppBar: appbar_1.MuiAppBar,
        MuiDrawer: drawer_1.MuiDrawer,
        MuiSlider: slider_1.MuiSlider,
        MuiFab: fab_1.MuiFab
    }
});
exports.default = exports.basicThemeV6;
