// @oclc/dist/themes/components/radio.js
import tokenData from '../../design-tokens/base-tokens.json';
const MuiRadio = {
    defaultProps: {
        disableRipple: true
    },
    styleOverrides: {
        root: {
            paddingLeft: 3,
            paddingRight: 3,
            padding: 3,
            color: tokenData.tokens.palette.grey.grey500,
            '&:hover': {
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineColor: tokenData.tokens.palette.blue.blue700
            },
            '&.Mui-focusVisible': {
                outlineStyle: 'solid',
                outlineWidth: '2px',
                outlineOffset: '1px',
                outlineColor: tokenData.tokens.palette.text.text300,
                boxShadow: `0px 0px 0px 2px ${tokenData.tokens.palette.background.white}`
            },
            '&.Mui-disabled': {
                color: tokenData.tokens.palette.grey.disabled,
                pointerEvents: 'unset',
                cursor: 'not-allowed',
                '&:hover': {
                    outline: 'none',
                    backgroundColor: 'transparent'
                }
            }
        }
    }
};
export default MuiRadio;
